assets/js/modules/adsense/components/setup/v2/SetupUseSnippetSwitch.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * AdSense Setup UseSnippet Switch Component.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { useEffect } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { MODULES_ADSENSE } from '../../../datastore/constants';
+import { parseAccountID } from '../../../util/parsing';
+import { UseSnippetSwitch } from '../../common';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupUseSnippetSwitch() {
+	const originalUseSnippet = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getOriginalUseSnippet()
+	);
+	const existingTag = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getExistingTag()
+	);
+	const clientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+
+	const { setUseSnippet } = useDispatch( MODULES_ADSENSE );
+
+	const hasExistingTag = Boolean( existingTag );
+	useEffect( () => {
+		if ( hasExistingTag ) {
+			setUseSnippet( false );
```

Comment:
@hussain-t looks like we forgot one more thing here:

> The component should also include a useEffect that, if an existing tag was found, sets the useSnippet setting to false **and saves it right away** (unless of course that setting was already set to false before).

So, we need to save the settings if the current useSnippet setting is truthy and we change it to false.

Reply:
Thanks, @eugene-manuilov, that's a good catch.

Reply:
@hussain-t we also need to save settings, which means that we need to call the `saveSettings` action: https://github.com/google/site-kit-wp/blob/4c156b533f57135c49dc64ff1adf2c1e143906cb/assets/js/modules/adsense/components/common/UseSnippetSwitch.js#L67